### PR TITLE
[Mellanox] Support new-platform-api sfp part, including the following APIs

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**
Support new-platform-api sfp part, including the following APIs
get_reset_status
get_tx_disable_channel
get_lpmode
get_power_override
reset
set_lpmode
The following APIs keep unimplemented due to suspicious SDK/FW issues that unable to write on I2C bus
tx_disable
tx_disable_channel
set_power_override

**- How I did it**

**- How to verify it**

**- Description for the changelog**
[Mellanox] Support new-platform-api sfp part, including the following APIs

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 1]
```diff
@@ -982,13 +1119,32 @@ def get_tx_power(self):
...
+        meta = self._init_sx_meta_data()
+        meta.access_cmd = SXD_ACCESS_CMD_GET
+
+        rc = sxd_access_reg_pmaos(pmaos, meta, 1, None, None)
+        assert rc == SXD_STATUS_SUCCESS, "sxd_access_reg_pmaos failed, rc = %d" % rc
+
+        # Reset SFP
+        pmaos.rst = 1
+        meta.access_cmd = SXD_ACCESS_CMD_SET
+        rc = sxd_access_reg_pmaos(pmaos, meta, 1, None, None)
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: change to REGISTER_NUM

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 2]
```diff
@@ -982,13 +1119,32 @@ def get_tx_power(self):
...
+        if handle is None:
+            return None
+
+        # Get PMAOS
+        pmaos = ku_pmaos_reg()
+        pmaos.module = self.sdk_index
+        meta = self._init_sx_meta_data()
+        meta.access_cmd = SXD_ACCESS_CMD_GET
+
+        rc = sxd_access_reg_pmaos(pmaos, meta, 1, None, None)
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: change to REGISTER_NUM

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 3]
```diff
@@ -732,7 +842,23 @@ def get_lpmode(self):
...
         """
-        return NotImplementedError
+        if self._open_sdk():
+            # Get MCION
+            mcion = ku_mcion_reg()
+            mcion.module = self.sdk_index
+            meta = self._init_sx_meta_data()
+            meta.access_cmd = SXD_ACCESS_CMD_GET
+
+            rc = sxd_access_reg_mcion(mcion, meta, 1, None, None)
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: change to REGISTER_NUM

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 4]
```diff
@@ -982,13 +1119,32 @@ def get_tx_power(self):
...
         Returns:
             A boolean, True if successful, False if not
+
+        refer plugins/sfpreset.py
         """
-        return NotImplementedError
-    
+        handle = self._open_sdk()
+        if handle is None:
+            return None
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: return false?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 5]
```diff
@@ -741,8 +867,19 @@ def get_power_override(self):
         Returns:
             A Boolean, True if power-override is enabled, False if disabled
         """
-        return NotImplementedError
-    
+        if self.sfp_type == QSFP_TYPE:
+            offset = 0
+            sfpd_obj = sff8436Dom()
+            if sfpd_obj is None:
+                return None
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: return value?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 6]
```diff
@@ -732,7 +842,23 @@ def get_lpmode(self):
...
+            meta.access_cmd = SXD_ACCESS_CMD_GET
+
+            rc = sxd_access_reg_mcion(mcion, meta, 1, None, None)
+            assert rc == SXD_STATUS_SUCCESS, "sxd_access_reg_mcion failed, rc = %d" % rc
+
+            # Get low power mode status
+            lpm_mask = 1 << 8
+            lpm_status = (lpm_mask & mcion.module_status_bits) != 0
+
+            return lpm_status
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: boolean?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 7]
```diff
@@ -723,7 +826,14 @@ def get_tx_disable_channel(self):
...
         """
-        return NotImplementedError
+        tx_disable_list = self.get_tx_disable()
+        if tx_disable_list is None:
+            return 0
+        tx_disabled = 0
+        for i in range(len(tx_disable_list)):
+            if tx_disable_list[i]:
+                tx_disabled |= 1 << i
+        return tx_disabled
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: boolean?

Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: no. it's sth like a bit string.

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 6 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: sure. it is a boolean.

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 5 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 4 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 3 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 2 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 1 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: fixed

**Pull request closed without merging**